### PR TITLE
feat: Implement typing indicator theme-v2 #95

### LINKED ISSUE #95: Typing indicator component
# SDK support
- https://github.com/GetStream/stream-chat-angular/issues/281
- https://github.com/GetStream/stream-chat-react/issues/1584

### PR DESCRIPTION
**Pull request by szuperaz**
### 🎯 Goal

_Describe why we are making this change_

### 🛠 Implementation details

_Provide a description of the implementation_

### 🎨 UI Changes

https://github.com/GetStream/stream-chat-angular/pull/315

closes #95

### CODE REVIEW

**Review by petyosi — APPROVED** [review]

File: docs/theming/component-variables.mdx [hunk 1]
```diff
@@ -158,6 +158,23 @@ Defined in: [https://github.com/GetStream/stream-chat-css/blob/main/src/v2/style
...
+| --------------------------------------------------- | ------------------------------------------------------------- |
+| `--str-chat__typing-indicator-font-family`          | The font used in the component                                |
+| `--str-chat__typing-indicator-color`                | The text/icon color of the component                          |
+| `--str-chat__typing-indicator-background-color`     | The background color of the component                         |
+| `--str-chat__typing-indicator-border-radius`        | The border radius used for the borders of the component       |
+| `--str-chat__typing-indicator-border-block-start`   | Top border of the component                                   |
+| `--str-chat__typing-indicator-border-block-end`     | Bottom border of the component                                |
+| `--str-chat__typing-indicator-border-inline-start`  | Left (right in RTL layout) border of the component            |
+| `--str-chat__typing-indicator-border-inline-start`  | Right (lrft in RTL layout) border of the component            |
```
Comment (oliverlaz) [inline docs/theming/component-variables.mdx:172]:
```suggestion
| `--str-chat__typing-indicator-border-inline-end`  | Right (left in RTL layout) border of the component            |
```

File: src/v2/styles/TypingIndicator/TypingIndicator-theme.scss [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+  --str-chat__typing-indicator-font-family: var(--str-chat__font-family);
+  --str-chat__typing-indicator-border-radius: none;
+  --str-chat__typing-indicator-color: var(--str-chat__text-low-emphasis-color);
+  --str-chat__typing-indicator-background-color: var(--str-chat__overlay-color);
+  --str-chat__typing-indicator-border-block-start: none;
+  --str-chat__typing-indicator-border-block-end: none;
+  --str-chat__typing-indicator-border-inline-start: none;
+  --str-chat__typing-indicator-border-inline-end: none;
+  --str-chat__typing-indicator-box-shadow: none;
+  --str-chat__typing-indicator-dot-background-color: var(--str-chat__text-color);
```
Comment (oliverlaz) [inline src/v2/styles/TypingIndicator/TypingIndicator-theme.scss:13]:
```suggestion
  --str-chat__typing-indicator__dot-background-color: var(--str-chat__text-color);
```

Nitpick: I'm not sure how closely BEM is followed across other files but, I think `dot` represents an element. This just caught my eye :)

**Review by oliverlaz — APPROVED** [review]

File: src/v2/styles/TypingIndicator/TypingIndicator-theme.scss (hunk 2 above)
Comment (szuperaz) [reply src/v2/styles/TypingIndicator/TypingIndicator-theme.scss:13]: In theme-v2 for CSS variables we use camel-case after the `str-chat__` prefix

File: docs/theming/component-variables.mdx (hunk 1 above)
Comment (szuperaz) [reply docs/theming/component-variables.mdx:172]: Will fix this after the ongoing PRs merged into theming-v2

File: src/v2/styles/TypingIndicator/TypingIndicator-theme.scss (hunk 2 above)
Comment (petyosi) [reply src/v2/styles/TypingIndicator/TypingIndicator-theme.scss:13]: I think BEM is just for class names, but I'm happy to be corrected. Let's stick to snake-case for CSS variables.

Comment (szuperaz) [reply src/v2/styles/TypingIndicator/TypingIndicator-theme.scss:13]: Unfortunately, our CSS class names are not 100% consistent, it's mostly BEM, but there are exceptions, the only consistent part is the `str-chat__` prefix.

Comment (szuperaz) [reply src/v2/styles/TypingIndicator/TypingIndicator-theme.scss:13]: https://github.com/GetStream/stream-chat-css/issues/135 created an issue to create some lint rules to check class and variable names https://github.com/GetStream/stream-chat-css/issues/135

**Pull request merged**